src/state_machine/signer/mod.rs

```diff
@@ -313,10 +341,20 @@ impl<SignerType: SignerTrait> Signer<SignerType> {
             Message::DkgPrivateBegin(dkg_private_begin) => {
                 self.dkg_private_begin(dkg_private_begin, rng)
             }
-            Message::DkgEndBegin(dkg_end_begin) => self.dkg_end_begin(dkg_end_begin),
-            Message::DkgPublicShares(dkg_public_shares) => self.dkg_public_share(dkg_public_shares),
+            Message::DkgEndBegin(dkg_end_begin) => self.dkg_end_begin(dkg_end_begin, rng),
+            Message::DkgPublicShares(dkg_public_shares) => {
+                if !self.embed_public_private_shares {
+                    self.dkg_public_share(dkg_public_shares)
+                } else {
+                    Ok(vec![])
```

Comment:
Would this warrant a warning if we are getting a dkgpubilcshares message? Or do we still expect them to send the DkgPublicShares, but we just wait for DkgPrivateShares to actually do anything with them?

Reply:
Yeah, the signers will still see the public and private shares coming from other signers, they just ignore them if configured to wait for the coordinator messages that embed them.